File: IKeep.Lib/Models/Floor.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace IKeep.Lib.Models
{
    public class Floor : Entity
    {
        public string Ref { get; set; }
        public string Name { get; set; }
        public Guid BuildingId { get; set; }

        [JsonIgnore]
        public ICollection<Area> Areas { get; set; }
        public List<Guid> AreasIds
        {
            get
            {
                return Areas == null ? new List<Guid>() : Areas.Select(x => x.Id).ToList();
            }
        }

        [JsonIgnore]
        public ICollection<Map> Maps { get; set; }
        public List<Guid> MapsIds
        {
            get
            {
                return Maps == null ? new List<Guid>() : Maps.Select(x => x.Id).ToList();
            }
        }
    }
}

File: IKeep.Web/ClientApp/src/models/Floor.js
import Entity from './Entity'

class Floor extends Entity
{
    constructor(json)
    {
        super(json);
        if(json)
        {
            this.Ref = json.ref; 
            this.Name = json.name;
            this.BuildingId = json.buildingId;
            this.AreasIds = json.areasIds;
            this.MapsIds = json.mapsIds;
        }
        else
        {
            this.Ref = ""; 
            this.Name = "";
            this.BuildingId = "";
            this.AreasIds = [];
            this.MapsIds = [];
        }
    }
}

export default Floor;